core/src/main/java/org/apache/accumulo/core/clientImpl/TableOperationsImpl.java

```diff
@@ -2145,4 +2150,37 @@ public void setTabletHostingGoal(String tableName, Range range, TabletHostingGoa
     }
   }
 
+  @Override
+  public Stream<HostingGoalForTablet> getTabletHostingGoal(final String tableName,
+      final Range range) throws TableNotFoundException {
+    EXISTING_TABLE_NAME.validate(tableName);
+
+    List<HostingGoalForTablet> hostingInfo = new ArrayList<>();
+
+    final Text scanRangeStart = (range.getStartKey() == null) ? null : range.getStartKey().getRow();
+    TableId tableId = context.getTableId(tableName);
+
+    try (TabletsMetadata m = context.getAmple().readTablets().forTable(tableId)
+        .overlapping(scanRangeStart, true, null).build()) {
```

Comment:
Would be good to narrow the data returned. Also the checkConsistency() call will automatically deal with concurrent splits.  However for concurrent merges it may throw an exceptions.  With this call the user gets a set of contiguous tablets or an exception if that can not be achieved.

```suggestion
        .overlapping(scanRangeStart, true, null).fetch(HOSTING_GOAL, PREV_ROW).checkConsistency().build()) {
```